qutebrowser/browser/webkit/webkithistory.py

```diff
@@ -48,7 +50,8 @@ def historyContains(self, url_string):
         Return:
             True if the url is in the history, False otherwise.
         """
-        return url_string in self._history.history_dict
+        with debug.log_time('sql', 'historyContains'):
+            return url_string in self._history
```

Comment:
If it really holds up, you might just decorate this method with [functools.lru_cache](https://docs.python.org/3/library/functools.html?highlight=lru_cache#functools.lru_cache) and tune the `maxsize` parameter. Though I'd expect that webkit does its own caching under the hood...